### ISSUE TITLE
[java] UselessParentheses false positive with boolean operators

### ISSUE THREAD

**Issue by xuanuy** [pmd/pmd#1918]
**Affects PMD Version:** 3.12.0

**Rule:** [UselessParentheses](https://pmd.github.io/latest/pmd_rules_java_codestyle.html#uselessparentheses)

**Description:**
I have the condition if(a || b || (c & d) || e)
Then I get Useless parentheses violation message.

**Code Sample demonstrating the issue:**

```java
if(a || b || (c & d) || e) {
 //do sth
}
```

**Running PMD through:** *Maven*

Java JDK 8

**Comment by AnthonyKot** [comment]
Sounds reasonable:
https://softwareengineering.stackexchange.com/questions/201175/should-i-use-parentheses-in-logical-statements-even-where-not-necessary
I will try to fix it

**Comment by AnthonyKot** [comment]
https://github.com/pmd/pmd/pull/1920

**Comment by adangel** [comment]
Since `&` has a higher precedence than `||`, the rule correct in saying: The parentheses are not needed. See [operator precedence](https://docs.oracle.com/javase/tutorial/java/nutsandbolts/operators.html).

However, in terms of readability, it's maybe better to allow these parentheses.
Similar discussion is here: #1673 

Maybe we should give this rule a property to enable/disable something like a "strict" mode?

**Comment by adangel** [comment]
Fixed with PMD 7

**Comment by oowekyala** [comment]
@blacelle PMD 7 is still in development, you can get daily snapshots from [sourceforge](https://sourceforge.net/projects/pmd/files/pmd/7.0.0-SNAPSHOT/) and the sonatype snapshot repository.

The bug has been fixed in the PMD 7 development branch, but may still be present in PMD 6. I'll reopen the issue with the proper label (we didn't have this procedure yet when this bug was closed).

**Comment by adangel** [comment]
This has been fixed with PMD 7.0.0-rc1.
